precompiles/dapps-staking/src/lib.rs

```diff
@@ -0,0 +1,231 @@
+//! Astar dApps staking interface.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use codec::{Decode, Encode};
+use fp_evm::{Context, ExitError, ExitSucceed, PrecompileOutput};
+use frame_support::dispatch::{Dispatchable, GetDispatchInfo, PostDispatchInfo};
+use frame_support::traits::Get;
+use pallet_evm::{AddressMapping, GasWeightMapping, Precompile};
+use sp_core::H160;
+use sp_runtime::traits::{SaturatedConversion, Zero};
+use sp_std::convert::TryInto;
+use sp_std::{marker::PhantomData, vec::Vec};
+extern crate alloc;
+
+mod utils;
+pub use utils::*;
+#[cfg(test)]
+mod mock;
+#[cfg(test)]
+mod tests;
+
+pub struct DappsStakingWrapper<R>(PhantomData<R>);
+
+impl<R> DappsStakingWrapper<R>
+where
+    R: pallet_evm::Config + pallet_dapps_staking::Config,
+    R::Call: From<pallet_dapps_staking::Call<R>>,
+{
+    /// Fetch current era from CurrentEra storage map
+    fn current_era() -> Result<PrecompileOutput, ExitError> {
+        let current_era = pallet_dapps_staking::CurrentEra::<R>::get();
+        let gas_used = R::GasWeightMapping::weight_to_gas(R::DbWeight::get().read);
+        println!(
+            "--- precompile DappsStaking response: current_era era={:?} gas_used={:?}",
+            current_era, gas_used
+        );
+
+        let output = utils::argument_from_u32(current_era);
+
+        Ok(PrecompileOutput {
+            exit_status: ExitSucceed::Returned,
+            cost: gas_used,
+            output,
+            logs: Default::default(),
+        })
+    }
+
+    /// Fetch reward and stake from EraRewardsAndStakes storage map
+    fn era_reward_and_stake(input: EvmInArg) -> Result<PrecompileOutput, ExitError> {
+        input.expecting_arguments(1).map_err(|e| exit_error(e))?;
+        let era = input.to_u256(1).low_u32();
+        let reward_and_stake = pallet_dapps_staking::EraRewardsAndStakes::<R>::get(era);
+        let (reward, staked) = if let Some(r) = reward_and_stake {
+            (r.rewards, r.staked)
+        } else {
+            (Zero::zero(), Zero::zero())
+        };
```

Comment:
I think you can just do: reward_and_stake.map_or(...)

Reply:
thanks! done